internal/util/util.go

```diff
@@ -60,15 +65,51 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	if opts.Type == "image" {
+		if opts.Attrs["push"] != "true" {
+			return nil
+		}
+
+		image := opts.Attrs["name"]
+		destSpec := opts.Attrs["name"]
+		dest, err := alltransports.ParseImageName(destSpec)
```

Comment:
I am not sure this code is correct, or necessary.  I know that @mtrmac and @vrothberg frown on parsing specs outside of github.com/containers/image or github.com/containers/common/libimage

Reply:
Yes; as a general principle I think the CLI layer, or something as close to it as possible, should be responsible for turning strings into a unique `types.ImageReference`.

Admittedly that’s not always possible, but it’s the thing to strive for.

Reply:
Does it help if it's moved to util? https://github.com/containers/buildah/pull/4677#discussion_r1176042766